src/templates/tags.js

```diff
@@ -3,19 +3,21 @@ import { Helmet } from "react-helmet";
 import { Link, graphql } from "gatsby";
 import Layout from "../components/Layout";
 
-class TagRoute extends React.Component {
-  render() {
-    const posts = this.props.data.allMarkdownRemark.edges;
+const TagRoute = (props) =>  {
+
+    const posts = props.data.allMarkdownRemark.edges;
+
     const postLinks = posts.map((post) => (
       <li key={post.node.fields.slug}>
         <Link to={post.node.fields.slug}>
           <h2 className="is-size-2">{post.node.frontmatter.title}</h2>
         </Link>
       </li>
     ));
-    const tag = this.props.pageContext.tag;
-    const title = this.props.data.site.siteMetadata.title;
-    const totalCount = this.props.data.allMarkdownRemark.totalCount;
+
+    const { tag } = props.pageContext;
+    const { title } = props.data.site.siteMetadata;
+    const { totalCount } = props.data.allMarkdownRemark;
```

Comment:
Same as the other component, what would you think about destructuring to using `{tag, title, totalCount, data}` so we're more aware of which props we're actually using?